libbeat/autodiscover/template/config.go

```diff
@@ -42,20 +42,20 @@ type MapperSettings []*struct {
 }
 
 // NewConfigMapper builds a template Mapper from given settings
-func NewConfigMapper(configs MapperSettings) (*Mapper, error) {
-	var mapper Mapper
+func NewConfigMapper(configs MapperSettings) (mapper Mapper, err error) {
```

Comment:
are you using the named return values? I think that's not needed from code

Reply:
It prevents me having to declare `mapper` on line 55 in the call to append.